### PR TITLE
Minor updates to onboarding

### PR DESCRIPTION
**Pull request by SheepTester**
# Info

adjustments to onboarding

## Changes

- replaced temporary images on first page
- made leaderboard look better on mobile
- whether onboarding was viewed is now stored on backend

# Type of Change

- [x] Bug Fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
      expected)
- [ ] Logistics Change (A change to a README, description, or dev workflow setup like
      linting/formatting)
- [ ] Continuous Integration Change (Related to deployment steps or continuous integration
      workflows)
- [ ] Other: (Fill In) 

# Testing

I have tested that my changes fully resolve the linked issue ...

- [x] locally on Desktop.
- [ ] on the live deployment preview on Desktop.
- [ ] on the live deployment preview on Mobile.
- [ ] I have added new Cypress tests that are passing.

# Checklist

- [ ] I have performed a self-review of my own code.
- [ ] I have followed the style guidelines of this project.
- [ ] I have documented any new functions in `/src/lib/*` and commented hard to understand areas
      anywhere else.
- [ ] My changes produce no new warnings.

# Screenshots

image

https://github.com/user-attachments/assets/b8822313-bae6-4aa1-aff6-2e6bf40c6e5f

image

### CODE REVIEW

**Review by newracket — APPROVED** [review]
i don't really know how animations work but it looks good to me

File: src/pages/index.tsx [hunk 1]
```diff
@@ -99,11 +99,7 @@ const PortalHomePage = ({
           // Start onboarding after checking in
           // TEMP: This should be saved server-side in the future
           // Do not start onboarding if user already attended other events
```
Comment (alexzhang1618) [inline src/pages/index.tsx]: We can remove these comments now right?

File: src/components/onboarding/Intro/index.tsx [hunk 2]
```diff
@@ -1,27 +1,155 @@
...
+
+// https://www.figma.com/design/GiWZdbzJ2uxyknpCrB9UK6/acm-onboarding
+const images: IntroImage[] = [
+  {
+    src: FocusPerson,
+    alt: 'A student smiling in a crowd',
+    desktopSize: [89, 46, 168, 168],
+    mobileSize: [24, 326, 109, 109],
+    round: true,
+  },
```
Comment (alexzhang1618) [inline src/components/onboarding/Intro/index.tsx]: nit: Let's make these typed so it's easier to read what number corresponds to what

File: src/components/onboarding/Intro/index.tsx [hunk 3]
```diff
@@ -1,27 +1,155 @@
...
+                left: `calc(${mobileSize[0] - MOBILE_OFFSET_X}px + ${
+                  (mobileSize[0] - MOBILE_OFFSET_X + mobileSize[2] / 2) * 0.2
+                }vw)`,
+                top: `${mobileSize[1] - MOBILE_OFFSET_Y}px`,
+                transformOrigin: `${MOBILE_OFFSET_X - mobileSize[0]}px ${
+                  MOBILE_OFFSET_Y - mobileSize[1]
+                }px`,
+                animationDelay: `${i * 0.1 + 0.5}s`,
+                animationDuration: `${i * 0.05 + 1}s`,
+              }}
```
Comment (alexzhang1618) [inline src/components/onboarding/Intro/index.tsx]: This is fine, but let's try to refactor out what we can to make this more readable if possible. I see animation stuff is getting reused, and a lot of logic is basically the same except switching between mobile and desktop. 

Maybe a helper function that returns the args, then we can just spread them here?
i.e. `getImageArgs(i, size, offsetY, offsetX)`

File: src/components/onboarding/Intro/index.tsx [hunk 4]
```diff
@@ -1,27 +1,155 @@
...
+    desktopSize: [173, 116, 631, 308],
+    mobileSize: [12, 173, 287, 206],
+    round: true,
+  },
+];
+images.reverse();
+const DESKTOP_OFFSET_X = 173 + 631 / 2;
+const DESKTOP_OFFSET_Y = 116 + 308 / 2;
+const MOBILE_OFFSET_X = 25 + 261 / 2;
+const MOBILE_OFFSET_Y = 203 + 146 / 2;
```
Comment (alexzhang1618) [inline src/components/onboarding/Intro/index.tsx]: Add some comments here on 1. why we're reversing and 2. what the calculation here does. I think this makes sense to store as constants for readability but it's your call

File: src/components/onboarding/Intro/index.tsx [hunk 5]
```diff
@@ -1,27 +1,155 @@
-import TempImage from '@/public/assets/graphics/cat404.png';
-import Image from 'next/image';
+import Allocation from '@/public/assets/graphics/onboarding/ACM_Fall23Allocation-JustinLu.jpg';
+import HoldBoba from '@/public/assets/graphics/onboarding/ACM_Fall24BitByteInfo_1-JustinLu.jpg';
+import Murou from '@/public/assets/graphics/onboarding/ACM_Fall24BitByteInfo_2-JustinLu.jpg';
+import Bonfire from '@/public/assets/graphics/onboarding/ACM_Fall24Bonfire_1-JustinLu.jpg';
+import RaymondBack from '@/public/assets/graphics/onboarding/ACM_Fall24Bonfire_2-JustinLu.jpg';
+import TiltKickoff from '@/public/assets/graphics/onboarding/ACM_Fall24Kickoff_1-JustinLu.jpg';
+import FocusPerson from '@/public/assets/graphics/onboarding/ACM_Fall24Kickoff_2-JustinLu.jpg';
+import KickoffBig from '@/public/assets/graphics/onboarding/ACM_Fall24Kickoff_3-JustinLu.jpg';
```
Comment (alexzhang1618) [inline src/components/onboarding/Intro/index.tsx]: Let's try and make these names as distinctive/descriptive as possible. I think
1. BitByteAllocation
2. BitByteInfoSession
3. BitByteSpeedFriending
4. BonfireMarshmallows
5. BonfireBeach (also I think this is Matthew, not Raymond) 👻 
6. KickoffSideView
7. KickoffCloseUp
8. KickoffFrontView

**Review by alexzhang1618 — APPROVED** [review]
Approving to unblock, some nits:

Let's try and rename the asset files to be as terse as possible. I think it's implicit that this is ACM, and we can probably shorten each file name to be something like
FA23_Allocation, FA24_BitByteInfo_1, etc.

File: src/components/onboarding/Intro/index.tsx (hunk 2 above)
Comment (SheepTester) [reply src/components/onboarding/Intro/index.tsx]: I'm assuming you mean to change this into an object with named properties rather than a tuple

**Pull request merged**